lib/index.js

```diff
@@ -46,8 +46,19 @@ export async function initializeServiceAgent({serviceType} = {}) {
       }
     }
 
-    // service agent not found, try to create it
-    const serviceAgent = await serviceAgents.generate({serviceType});
+    let serviceAgent;
+    try {
+      // service agent not found, try to create it
+      serviceAgent = await serviceAgents.generate({serviceType});
+    } catch(e) {
+      if(e.name !== 'InvalidStateError') {
+        // some non-recoverable failure, bail
+        throw e;
+      }
+      // invalid state error, retry
```

Comment:
If an `InvalidStateError` is thrown, is there a guarantee that it eventually won't be thrown again? If not, we'll be stuck in this loop forever. Can it result from a bad configuration -- such that we would never leave the loop?

If we're not certain that the `InvalidStateError` will eventually stop, we should add a timer and check it before continuing, if the timer has expired, we should throw a timeout error so we break out of the loop.

Reply:
If we need the timeout, the pattern is:

```js
const signal = AbortSignal.timeout(2000);
while(true) {
  // ...

  if(signal.aborted) {
    throw new BedrockError(..., {name: 'TimeoutError', ...});
  }
}
```

Reply:
@dlongley a loop might be OK as long as there is appropriate logging around the condition.  But if we can arrange it so that there is a `bedrock.exit` after some number of failed retries or some time limit as you said, that's fine to.  We just need to make sure that operations people have good guidance on what to do if the container fails to start up.

Reply:
Addressed https://github.com/digitalbazaar/bedrock-service-agent/pull/23/commits/65d373d490ce0d0b078a5dd55404153fc3750420